### ISSUE TITLE
Unexpected Core Diffracted phases

### ISSUE THREAD

**Issue by benjaminfernando** [crotwell/TauP#8]
Hi, 

I'm using taup_curve to generate travel-time curves for a Poisson solid model of the Sun. 

The travel times look fine, except for the occurrence of Pdiff and Sdiff phases (at 120 degrees and 180 degrees only), with exactly the same travel time. I can't see that there would be anything physical for these phases to diffract off of (there are no discontinuities, only gradual transitions). 

Do you have any idea why this might be? 

Files attached (I have saved the .tvel as a .txt because I cannot upload .tvel).

Thanks,
Ben

[solar_constant_density_backup copy.txt](https://github.com/crotwell/TauP/files/1800937/solar_constant_density_backup.copy.txt)

[solartt.pdf](https://github.com/crotwell/TauP/files/1800938/solartt.pdf)

**Comment by crotwell** [comment]
The diffracted phases in TauP are a bit of a kludge as they don't strictly exist in the ray theoretic calculations. In your model, because there is not a "core" delineated by a first order discontinuity in velocity, TauP puts it at the center of the model, ie a core of radius 0, then calculates  the diffraction at this fake boundary, which is wrong.

I will put in code to disallow Pdiff and Sdiff if the core has zero radius, but in the mean time you could just change your phase list to not include Pdiff or Sdiff, so use just maybe `-ph p,s,P,S` instead of the default. Obviously add any others you need. See section 4 for how phases are named.

Let me know if you have problems...

**Comment by crotwell** [comment]
Also, given your solar model is so different from the earth, many of the defaults may not be appriate. In particular, TauP may generate an overly densely sampled model. These properties might be worth exploring to see if you can get away with coarser values that allow the calculations to be much quicker, depending on what accuracy you need. See section 3.1 for more info on the properties. For example:

```
# max and min delta p in seconds per radian
taup.create.minDeltaP=0.1
taup.create.maxDeltaP=100.0

# max depth interval in kilometers
taup.create.maxDepthInterval=1000.0

# max range interval in degrees
taup.create.maxRangeInterval=2.5

# max estimate of error due to linear interpolation, in seconds
taup.create.maxInterpError=10.0
```

**Comment by crotwell** [comment]
Fixed in commit f74aadf578a962c100048ede339ef70c88f098d8
